prometheus/test_assets_test.go

```diff
@@ -0,0 +1,53 @@
+package prometheus
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/prometheus/common/model"
+
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/stretchr/testify/mock"
+)
+
+type flag int
+
+type MockAPI struct {
+	flag
+	v1.API
+	mock.Mock
+}
+
+// Override the QueryRange method with a mock implementation
+func (m *MockAPI) QueryRange(context.Context, string, v1.Range, ...v1.Option) (model.Value, v1.Warnings, error) {
+	if m.flag == 1 {
+		samples := []model.SamplePair{
+			{Value: 1, Timestamp: model.TimeFromUnix(1)},
+			{Value: 2, Timestamp: model.TimeFromUnix(2)},
+			{Value: 3, Timestamp: model.TimeFromUnix(3)},
+		}
+
+		// Create a SampleStream
+		stream := model.SampleStream{
+			Metric: model.Metric{},
+			Values: samples,
+		}
+
+		// Wrap the SampleStream in a Matrix
+		matrix := model.Matrix{&stream}
+		return matrix, nil, nil
+	}
+	if m.flag == 2 {
+		return model.Matrix{}, nil, fmt.Errorf("sample error")
+	}
+	return nil, nil, nil
+}
+
+func (m *MockAPI) Query(ctx context.Context, query string, ts time.Time, opts ...v1.Option) (model.Value, v1.Warnings, error) {
```

Comment:
Why is this mock implementation always returning `nil`. Can you add some success cases as well?

Reply:
The reason why we are returning nil for the mock implementation is to mock the function as no error is being produced. The error produced case i.e. the success case which is already being taken care when wrong or mock information is passed through the Query func which raises the error when called, hence the error guard clauses are being covered.